ykbh/src/lib.rs

```diff
@@ -0,0 +1,310 @@
+#![feature(test)]
+extern crate test;
+
+use ykpack::{self, Local, Terminator, Statement, IPlace, Constant, ConstantInt, UnsignedInt};
+use yktrace::sir::SIR;
+use std::convert::TryInto;
+
+const ZST: Object = Object { data: Vec::new() };
+
+#[derive(Clone, Debug)]
+/// A data object, storing bytes.
+pub struct Object {
+    data: Vec<u8>
+}
+
+impl Object {
+
+    /// Creates a new Object from a `usize`.
+    fn from_usize(ptr: usize) -> Object {
+        let data = ptr.to_be_bytes();
+        Object {
+            data: Vec::from(data)
+        }
+    }
+
+    /// Creates a new Object from a `u8`.
+    fn from_u8(val: u8) -> Object {
+        Object {
+            data: Vec::from([val])
+        }
+    }
+
+    /// Converts the objects data back to a `usize`.
+    fn to_usize(&self) -> usize {
+        usize::from_be_bytes(self.data.as_slice().try_into().expect("Unexpected data length."))
```

Comment:
`be` means big endian? is there a way of asking for native endianess?

Reply:
Yeah, we should probably do that.

Reply:
```suggestion
        usize::from_ne_bytes(self.data.as_slice().try_into().expect("Unexpected data length."))
```